### PR TITLE
Creation of Registration company Interface and Create views for Companies

### LINKED ISSUE #33: Create views for Companies
We're going to create the views for the companies controller

### LINKED ISSUE #34: Creation of Registration company Interface
Creation of the interface with fields name, address

### PR DESCRIPTION
**Pull request by SEVERO96**
CRUD of the module of the branches

### CODE REVIEW

File: checador/app/assets/stylesheets/companies.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the companies controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (YilKanda) [inline checador/app/assets/stylesheets/companies.scss]: If you don't occupy this file, delete it.

File: checador/app/helpers/companies_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module CompaniesHelper
+end
```
Comment (YilKanda) [inline checador/app/helpers/companies_helper.rb]: If you don't occupy this file, delete it.

File: checador/spec/helpers/companies_helper_spec.rb [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+# Specs in this file have access to a helper object that includes
+# the CompaniesHelper. For example:
+#
+# describe CompaniesHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
```
Comment (YilKanda) [inline checador/spec/helpers/companies_helper_spec.rb]: If you do not occupy this code, delete it.

**Review by YilKanda — COMMENTED** [review]
As a suggestion, be more specific with your commit and your PR, you explain that you did the CRUD of the branches, but how does that change benefit the project? Why did it need to be implemented? This can provide a general idea of ​​what was done in your PR.

File: checador/app/models/company.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+class Company < ApplicationRecord
```
Comment (jclopezdev) [inline checador/app/models/company.rb:1]: What do you think of adding some validations of the model here? I want to think that you don't need a Company without a `name`, and also add some specs for this model

File: checador/db/migrate/20210723044029_create_companies.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class CreateCompanies < ActiveRecord::Migration[6.1]
+  def change
+    create_table :companies do |t|
+      t.string :name
+      t.string :addres
```
Comment (jclopezdev) [inline checador/db/migrate/20210723044029_create_companies.rb:5]: You have a `typo` here, I think you mean `address`

**Review by jclopezdev — CHANGES_REQUESTED** [review]

File: checador/app/models/company.rb (hunk 4 above)
Comment (ManuelAcostaGuerrero) [reply checador/app/models/company.rb:1]: Hello, the validations are intended to be generated by having the modules fully integrated. For now we think we better leave it like this

File: checador/db/migrate/20210723044029_create_companies.rb (hunk 5 above)
Comment (ManuelAcostaGuerrero) [reply checador/db/migrate/20210723044029_create_companies.rb:5]: Thanks, the changes were made, could you check

File: checador/spec/helpers/companies_helper_spec.rb (hunk 3 above)
Comment (ManuelAcostaGuerrero) [reply checador/spec/helpers/companies_helper_spec.rb]: Thanks, the changes were made, could you check

File: checador/app/helpers/companies_helper.rb (hunk 2 above)
Comment (ManuelAcostaGuerrero) [reply checador/app/helpers/companies_helper.rb]: Thanks, the changes were made, could you check

File: checador/app/assets/stylesheets/companies.scss (hunk 1 above)
Comment (ManuelAcostaGuerrero) [reply checador/app/assets/stylesheets/companies.scss]: Thanks, the changes were made, could you check

**Review by ManuelAcostaGuerrero — DISMISSED** [review]

File: checador/app/views/companies/index.html.erb [hunk 6]
```diff
@@ -1 +1,30 @@
...
+
+<p id="notice"><%= notice %></p>
+
+<h1>Companies</h1>
+
+<table>
+  <thead>
+    <tr>
+      <th>Name</th>
+      <th>Addres</th>
```
Comment (jclopezdev) [inline checador/app/views/companies/index.html.erb]: You also need to update the `Addres` word here

File: checador/app/controllers/companies_controller.rb [hunk 7]
```diff
@@ -1,41 +1,71 @@
...
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_company
+      @company = Company.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def company_params
+      params.require(:company).permit(:name, :addres)
```
Comment (jclopezdev) [inline checador/app/controllers/companies_controller.rb]: And here...

File: checador/app/views/companies/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= form.label :name %>
+    <%= form.text_field :name %>
+  </div>
+
+  <div class="field">
+    <%= form.label :addres %>
```
Comment (jclopezdev) [inline checador/app/views/companies/_form.html.erb]: And also here...

**Review by jclopezdev — CHANGES_REQUESTED** [review]
I can still see a lot of typos with the `address` word, please take a look to fix it in all the files.

File: checador/app/views/companies/_form.html.erb (hunk 8 above)
Comment (xFerchoVJ) [reply checador/app/views/companies/_form.html.erb]: Thank you, Juan Carlos!
We're going to change it!

Comment (SEVERO96) [reply checador/app/views/companies/_form.html.erb]: ok

File: checador/app/views/companies/index.html.erb (hunk 6 above)
Comment (ManuelAcostaGuerrero) [reply checador/app/views/companies/index.html.erb]: Ready Thank you very much we are attentive to any changes

File: checador/app/controllers/companies_controller.rb (hunk 7 above)
Comment (ManuelAcostaGuerrero) [reply checador/app/controllers/companies_controller.rb]: Ready Thank you very much we are attentive to any changes

File: checador/app/views/companies/_form.html.erb (hunk 8 above)
Comment (ManuelAcostaGuerrero) [reply checador/app/views/companies/_form.html.erb]: Ready Thank you very much we are attentive to any changes

**Review by jairolan — APPROVED** [review]

**Review by YilKanda — APPROVED** [review]
LGTM!

**Review by xFerchoVJ — APPROVED** [review]
LGTM

**Pull request merged**